### DISCUSSION TITLE
MeshCentral Paid Freelancer Help ?

### DISCUSSION THREAD

**Discussion by compumatter** [Ylianst/MeshCentral#4983] [General]
I have just posted a job on Upwork.com and thought I'd check with the community directly to see if anyone is interested (directly or through upwork.com - either way).  

I have managed to get MeshCentral installed but have ended up with only internal / LAN working  and even then only those Pro Windows with RDP built in.  I can't get agents working on LAN or WAN.  

**This is my job post if anyone is interested.**

We need someone who is knowledgeable about configuring the schema features applicable to us as extracted from here https://github.com/Ylianst/MeshCentral/blob/master/meshcentral-config-schema.json

We need someone knowledgeable (who has used MeshCentral regularly)

We do not want to "Learn Together".  We need you to be the teacher.

Immediate Requirements:

* Review our meshcentral server installation on a Linux Ubuntu 22.04 Server and see if it needs improving.
* Help us to understand how to setup log files for the server and agent so we can troubleshoot problems.
* Show how to best configue remote access to any of the desktops on our Internal network from the outside world (OSx, Windows and Linux)
* Show us how to remote access desktops (OSx, Windows and Linux) in the outside world from our Internal network.  

Future Needs - not required now but of interest to us if you have any thoughts about this: 
* There is a page on improving security on Ubuntu server installs. Though it specifies Mongodb, we would like to use mariadb which we have been told is supported.  https://meshcentral-community.com/doku.php?id=howto:installation:manual:ubuntu:moresecure

Please describe to us your use of Meshcentral in the past so we may know you are knowledgeable about this product

**Comment by dinger1986** [comment]
I think my install script covers most of that. https://github.com/techahold/Meshcentral-Installer

I would suggest Debian though.

A lot of what you are asking I would say is basic and someone could help easily enough.

**Reply by thermionic** [reply]
@dinger1986  just to let you know I took apart your script and put it back together again altered for my use case with a couple of additions. It still needs more work, but is getting there https://github.com/thermionic/Meshcentral-Installer/tree/thermionic-2023-02-12 

Many thanks for writing it in the first place!

**Reply by dinger1986** [reply]
Yeah I saw that you had forked it.

Anything you think is with PRing happy to look at!

**Reply by compumatter** [reply]
@thermionic  @dinger1986 I'm a little confused about the insertion of the letsencrypt stuff.  If I already have a working wildcard cert in place with letsencrypt ie; *.mydomain.com should I be filling out the letsencrypt related stuff in the bash script - or is that just for when you do not have a letsencrypt cert and need one put in place?

**Reply by thermionic** [reply]
If you have a valid cert, then a letsencrypt cert isn't needed. The point about using letsencrypt is that one doesn't have to bother manually renewing the cert...

If you want to go "bleeding edge" then see what you can get working with 22.04, personally I prefer stability, so for the time being I;m staying on 20.04 with mongodb, however if you have a requirement to be on 22.04, then I'd suggest following the script from @dinger1986 and going with postgres.

**Reply by dinger1986** [reply]
I'm going to look to add MySQL/Mariadb to it as well

**Comment by compumatter** [comment]
Thanks.Looks interesting.  You might consider adding Mariadb / MySQL to that list as mongodb doesn't work with Ubuntu 22.04 without ssllib hacks .  We use mariadb / MySQL I'll have a look and report back

On Feb 14, 2023, at 12:17 AM, dinger1986 ***@***.***> wrote:﻿
I think my install script covers most of that. https://github.com/techahold/Meshcentral-Installer
I would suggest Debian though.
A lot of what you are asking I would say is basic and someone could help easily enough.

**Reply by dinger1986** [reply]
I always use postgres tbf but yes can look at adding mysql, should be simple enough

**Comment by compumatter** [comment]
Thanks again @dinger1986 for assisting me directly with this issue.  I'm up and running.

**Comment by matthiscock** [comment]
Are you stilll looking for assistance on this.
We run a corporate network on Mesh but also are building on top of it, using live chat systems, performance measuring and hopefully soon a new front end.

We run it under windows, linux and unriad based environments (with reverse proxies) along with docker :)

**Comment by si458** [comment]
happy to help! u can always msg me/email/tweet/tele/whats/disc me!
ill put it bluntly to u if ur doing something wrong haha

**Reply by Tutorsclubdesk** [reply]
Hello, I need someone to develop MC with extra features.

**Reply by si458** [reply]
@Tutorsclubdesk depending on what u want/need you can contact myself, my email address is on my github page or im si458 on tele/whats/disc/sig/tweet etc
